### PR TITLE
feat(content-answers): intelligence api

### PR DESCRIPTION
**Pull request by greg-in-a-box**

### CODE REVIEW

File: src/api/Intelligence.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @file Helper for the Box AI API endpoint
+ * @author Box
+ */
+
+import Base from './Base';
+import { ERROR_CODE_INTELLIGENCE } from '../constants';
+import type { BoxItem } from '../common/types/core';
+import type { ElementsErrorCallback } from '../common/types/api';
+
+class Intelligence extends Base {
```
Comment (megansmith-box) [inline src/api/Intelligence.js:12]: I see other recent Apis extend `MarkerBasedApi` which extends `Base` , do you think we'd ever need that functionality for Intelligence API?

File: src/api/Intelligence.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    /**
+     * @property {Function}
+     */
+    errorCallback: ElementsErrorCallback;
+
+    /**
+     * API endpoint to ask ai a question
+     *
+     * @param {string} prompt - Question
+     * @param {Array<object>} items - Array of items to ask about
```
Comment (megansmith-box) [inline src/api/Intelligence.js:27]: Are we missing the last 2 params here?

File: src/api/Intelligence.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        const body = {
+            data: {
+                mode: 'single_item_qa',
+                prompt,
+                items,
+            },
+        };
+
+        this.successCallback = successCallback;
+        this.errorCallback = errorCallback;
```
Comment (megansmith-box) [inline src/api/Intelligence.js:56]: Why do these need to be set as variables of the class?

Comment (megansmith-box) [reply src/api/Intelligence.js:56]: Should we also validate that successCallback and errorCallback are valid or have a default value?

File: src/api/Intelligence.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        await this.xhr
+            .post({
+                url,
+                id: items[0].id,
+                data: body,
+            })
+            .then(this.successCallback)
+            .catch(this.errorHandler);
```
Comment (megansmith-box) [inline src/api/Intelligence.js:65]: should this be errorCallback?

File: src/api/Intelligence.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            },
+        };
+
+        this.successCallback = successCallback;
+        this.errorCallback = errorCallback;
+
+        await this.xhr
+            .post({
+                url,
+                id: items[0].id,
```
Comment (megansmith-box) [inline src/api/Intelligence.js:61]: Why is id = to the first items id?

File: src/api/Intelligence.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                data: body,
```
Comment (megansmith-box) [inline src/api/Intelligence.js:62]: This seems kind of strange, should it be `data.data`?

File: src/api/Intelligence.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * API endpoint to ask ai a question
+     *
+     * @param {string} prompt - Question
+     * @param {Array<object>} items - Array of items to ask about
+     * @return {Promise}
+     */
+    async ask(prompt: string, items: Array<BoxItem>, successCallback: Function, errorCallback: ElementsErrorCallback) {
+        this.errorCode = ERROR_CODE_INTELLIGENCE;
```
Comment (megansmith-box) [inline src/api/Intelligence.js:31]: Why is this error code set here like this?

File: src/api/Intelligence.js (hunk 4 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:65]: the errorHandler calls errorCallback

File: src/api/Intelligence.js (hunk 3 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:56]: pattern is used in other clients, its been set for the base class to use

File: src/api/Intelligence.js (hunk 5 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:61]: I honestly dont know why it even needs the id, its a requirement of the the xhr.post, but the items array will always have one item for now till this client is used for multi. items is required by the intelligence request it handles multi only

File: src/api/Intelligence.js (hunk 7 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:31]: this.errorCode is used by the base class to be applied when the errorHandler is being used

File: src/api/Intelligence.js (hunk 1 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:12]: I dont think so

File: src/api/Intelligence.js (hunk 4 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:65]: if anything I think I should be using `successHandler`

File: src/api/Intelligence.js (hunk 3 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:56]: https://github.com/box/box-ui-elements/blob/master/src/api/Base.js#L182
https://github.com/box/box-ui-elements/blob/master/src/api/Base.js#L194
Base class is handling the validation of those callbacks

File: src/api/Intelligence.js (hunk 5 above)
Comment (greg-in-a-box) [reply src/api/Intelligence.js:56]: looked it up, it seems to be using the id to get a writetoken to do whatever it needs to do

Comment (TaoJiang98) [reply src/api/Intelligence.js:56]: yeah since the current mode we are using is 'single_item_qa', the array item will always has one item only.

**Review by TaoJiang98 — APPROVED** [review]
lgtm!

File: src/api/Intelligence.js (hunk 5 above)
Comment (megansmith-box) [reply src/api/Intelligence.js:56]: [nit] should we validate that items length is always 1?

**Review by megansmith-box — APPROVED** [review]

**Review by bfoxx1906 — APPROVED** [review]
previous reviews make sense

**Pull request merged**